src/__test__/dataparitytests/transactions.parity.test.ts

```diff
@@ -93,275 +33,125 @@ export function transactionTests (createClient: () => Promise<TestClient>) {
-                                outputs{
-                                  address
-                                  index
-                                  value
-                                }
-    
-                                totalOutput
-                                includedAt
-                              }
-                            }`
-      })
-
-      const restResultInputs = restResult['Right']['ctsInputs']
+      const restResultInputs = restData.ctsInputs
       let restResultValues = 0
 
       for (const input of restResultInputs) {
-        restResultValues += Number(input[1]['getCoin'])
+        restResultValues += Number(input[1].getCoin)
       }
 
-      const graphQLInputs = graphQLResult['data']['transactions'][0]['inputs']
+      const graphQLInputs = graphQLData.inputs
       let graphQLValues = 0
 
       for (const input of graphQLInputs) {
-        graphQLValues += Number(input['value'])
+        graphQLValues += Number(input.value)
       }
 
       expect(graphQLValues).toEqual(restResultValues)
     })
 
     it('return the correct Output Addresses', async () => {
-      const restResult = await getDataFromAPI('txs/summary/1ac36644733c367ee4c551413d799d2e395d6ddfe14bebf1c281e6e826901762')
-      const graphQLResult = await client.query({
-        query: gql`query TxById{
-                              transactions (where: {id: {_eq:"1ac36644733c367ee4c551413d799d2e395d6ddfe14bebf1c281e6e826901762"}})
-                              {
-                                id
-                                fee
-                                block{
-                                  number
-                                  createdAt
-                                }
-                                inputs(order_by: { index: asc }) {
-                                  address
-                                  value
-                                  sourceTxId
-                                  sourceTxIndex
-                                }
-                                outputs(order_by: { index: asc }){
-                                  address
-                                  index
-                                  value
-                                }
-
-                                totalOutput
-                                includedAt
-                              }
-                            }`
-      })
-
-      const restResultOutputs = restResult['Right']['ctsOutputs']
-      const restResultAddresses = []
+      const restResultOutputs = restData.ctsOutputs
+      const restResultAddresses = new Set()
 
       for (const output of restResultOutputs) {
-        restResultAddresses.push(output[0])
+        restResultAddresses.add(output[0])
       }
 
-      const graphQLOutputs = graphQLResult['data']['transactions'][0]['outputs']
-      const graphQLAddresses = []
+      const graphQLOutputs = graphQLData.outputs
+      const graphQLAddresses = new Set()
```

Comment:
This test was failing for me - elements in array were the same but the order was different so I used Set instead of Array.

Reply:
The test was failing because the GQL query was originally ordered. As ordering does matter, we should be using arrays instead of Sets. If you order the outputs by index this test will pass. 

```
                                inputs(order_by: { index: asc }) {
                                  address
                                  value
                                  sourceTxId
                                  sourceTxIndex
                                }
                                outputs(order_by: { index: asc }){
                                  address
                                  index
                                  value
```